tokio-postgres/src/connect.rs

```diff
@@ -448,17 +458,18 @@ async fn refresh(client: Client, config: &Config) {
         control_conn_host = socket_config.unwrap().hostname.unwrap();
     }
 
+    info!("Executing query: `select * from yb_servers()` to fetch list of servers");
+    let rows = client
+        .query("select * from yb_servers()", &[])
+        .await
+        .unwrap();
+
     let mut host_list = HOST_INFO.lock().unwrap();
     let mut failed_host_list = FAILED_HOSTS.lock().unwrap();
     let mut placement_info_map = PLACEMENT_INFO_MAP.lock().unwrap();
     let mut public_host_map = PUBLIC_HOST_MAP.lock().unwrap();
     let mut host_to_port_map = HOST_TO_PORT_MAP.lock().unwrap();
-    info!("Executing query: `select * from yb_servers()` to fetch list of servers");
-    for row in client
-        .query("select * from yb_servers()", &[])
-        .await
-        .unwrap()
-    {
+    for row in rows {
```

Comment:
The await in this code block would cause a deadlock with the group of mutex guards just above it. Moving the query to before the mutexes prevents the deadlock.